genesis/engine/solvers/rigid/collider_decomp.py

```diff
@@ -50,6 +51,42 @@ def rotmatx(matin, i0, i1, i2, f0, f1, f2):
     return matres
 
 
+@ti.func
+def check_collision_valid(
```

Comment:
can you keep it as a class method instead of a standalone function to minimize code change?

Reply:
I pulled them out because if we keep it as a class method, I should use `self_unused.check_collision_valid(...)`. Would it be okay to do in this way?